src/pages/SignUp.js

```diff
@@ -0,0 +1,80 @@
+import './SignUp.scss';
+import { Link } from 'react-router-dom';
+import { AGREEMENT } from './AGREEMENT';
+import { useEffect, useState } from 'react';
+import AgreementMap from '../components/AgreementMap';
+
+const SignUp = () => {
+  const [isAll, setIsAll] = useState(false);
+  const [isChecked, setIsChecked] = useState({ 1: false, 2: false, 3: false });
+
+  const handleChecked = e => {
+    const { id, checked } = e.target;
+    setIsChecked({ ...isChecked, [id]: checked });
+  };
+
+  const isAllChecked = Object.values(isChecked).filter(el => el === true);
+
+  const allChecked = e => {
+    const { checked } = e.target;
+    setIsAll(prev => !prev);
+
+    let checkedObj = {};
+    for (let i = 0; i <= AGREEMENT.length - 1; i++) {
+      checkedObj = { ...checkedObj, [AGREEMENT[i].id]: checked };
+    }
+    setIsChecked(checkedObj);
+  };
+
+  const isValid = isChecked[1] === true && isChecked[2] === true;
```

Comment:
isChecked 객체에 모든 key에 대한 value들이 Boolean 값으로 저장됩니다 
따라서
```suggestion
  const isValid = isChecked[1]  && isChecked[2] ;
```
위와같이 축약할 수 도 있습니다!